### PR TITLE
[php] Implement native iterator on arrays

### PR DESCRIPTION
**Pull request by benmerckx**
This one implements php's [`Iterator`](https://www.php.net/manual/en/class.iterator.php) on arrays using the internal array position. It eases interfacing with generated haxe code from the php side. (Eg. allows us to `foreach ($anyHaxeArray as $key => $item)`).

### CODE REVIEW

File: std/php/_std/Array.hx [hunk 1]
```diff
@@ -273,3 +298,12 @@ private extern interface ArrayAccess<K, V> {
 	private function offsetSet(offset:K, value:V):Void;
 	private function offsetUnset(offset:K):Void;
 }
+
+@:native('Iterator')
+private extern interface NativeIterator<K, V> {
```
Comment (RealyUniqueName) [inline std/php/_std/Array.hx]: This should be moved to an own module in `php` package similar to `php.ArrayAccess`. It could be useful in other cases

**Comment by RealyUniqueName** [comment]
Haxe's `Array` should implement Php's `IteratorAggregate` instead of `Iterator`, then this would be possible:
```php
foreach($haxeArray as $key => $value) {
    foreach($haxeArray as $key2 => $value2) {
    }
}
```

File: std/php/_std/Array.hx [hunk 2]
```diff
@@ -273,3 +278,8 @@ private extern interface ArrayAccess<K, V> {
 	private function offsetSet(offset:K, value:V):Void;
 	private function offsetUnset(offset:K):Void;
 }
+
+@:native('ArrayIterator')
+private extern class NativeArrayIterator<T> implements Traversable {
```
Comment (RealyUniqueName) [inline std/php/_std/Array.hx]: Why private to `Array` again? :)
This is a common PHP native class which could be used anywhere, not only in `Array`

Comment (benmerckx) [reply std/php/_std/Array.hx]: Well, it extends `SeekableIterator`, `Countable` and `Serializable`. Should I add those to php package too to implement it properly?

Comment (RealyUniqueName) [reply std/php/_std/Array.hx]: That would be great. Though, if you don't have time to, I can do it myself.

Comment (benmerckx) [reply std/php/_std/Array.hx]: I'll add them. Wasn't sure if we wanted all of those in there :)

Comment (RealyUniqueName) [reply std/php/_std/Array.hx]: We want externs for all of the PHP native types and functions eventually.

File: std/php/ArrayIterator.hx [hunk 3]
```diff
@@ -0,0 +1,57 @@
...
+ */
+
+package php;
+
+/**
+	@see https://www.php.net/manual/en/class.arrayiterator.php
+**/
+@:native('ArrayIterator')
+extern class ArrayIterator<K, V> implements php.ArrayAccess<K, V> implements SeekableIterator<K, V> implements Countable implements Serializable {
+	static final STD_PROP_LIST:Int;
```
Comment (RealyUniqueName) [inline std/php/ArrayIterator.hx]: These should have `@:phpClassConst` meta

**Review by RealyUniqueName — CHANGES_REQUESTED** [review]

**Review by RealyUniqueName — APPROVED** [review]

**Pull request merged**